firebase-firestore/src/main/java/com/google/firebase/firestore/PersistentCacheIndexManager.java

```diff
@@ -0,0 +1,55 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RestrictTo;
+import com.google.firebase.firestore.core.FirestoreClient;
+
+/**
+ * A {@code PersistentCacheIndexManager} which you can config persistent cache indexes used for
+ * local query execution.
+ *
+ * <p>To use, calling {@link FirebaseFirestore#getPersistentCacheIndexManager()} to get an instance.
+ */
+// TODO(csi): Remove the `hide` and scope annotations.
+/** @hide */
+@RestrictTo(RestrictTo.Scope.LIBRARY)
+public class PersistentCacheIndexManager {
+  @NonNull private FirestoreClient client;
+
+  @RestrictTo(RestrictTo.Scope.LIBRARY)
+  public PersistentCacheIndexManager(FirestoreClient client) {
```

Comment:
The reason for adding visibility modifier is because this will prevent developer from creating it directly. All PersistentCacheIndexManager object should comes from `getPersistentCacheIndexManager()`